[Issue] platform_test is failing on macOS darwin/amd64

<!--
Note: Please search to see if an issue already exists for the bug you encountered.
-->

Possible regression from #782.

### Current Behavior

```
ok      github.com/regclient/regclient/types/mediatype  6.842s  coverage: 100.0% of statements
--- FAIL: TestPlatformParse (0.00s)
    --- FAIL: TestPlatformParse/macos_amd64 (0.00s)
        platform_test.go:192: platform did not match, want darwin/amd64/v3, received darwin/amd64
FAIL
coverage: 84.1% of statements
FAIL    github.com/regclient/regclient/types/platform   6.729s
```

### Expected Behavior

<!-- A concise description of what you expected to happen. -->

The test should pass, and the make command should succeed.

### Steps To Reproduce

Build regclient using official instructions. Clone and make.

<!--
Example: steps to reproduce the behavior:
1. In this environment...
2. With this config...
3. Run '...'
4. See error...
-->

### Version

`0.7.2`

### Environment

<!--
Please include:
- How you are running the tool: downloaded binary or from within a container
- Your platform: Linux, Windows, or Mac, x86 or ARM.
- Your registry: Docker Hub, ECR, GCR, ACR, Harbor, registry:2, etc.
-->

- Running as binary or container: N/A, test failure.
- Host platform: `Darwin Kernel Version 23.6.0: Thu Sep 12 23:34:49 PDT 2024; root:xnu-10063.141.1.701.1~1/RELEASE_X86_64`
- Registry description: N/A

### Anything else

<!--
Links? References? Anything that will give us more context about the issue that you are encountering!
-->

--- a/types/platform/platform_test.go
+++ b/types/platform/platform_test.go
@@ -20,20 +20,11 @@ func TestPlatformParse(t *testing.T) {
 		winGoal.Variant = platLocal.Variant
 		winGoal.OSVersion = platLocal.OSVersion
 	}
-	linuxAMD64Goal := Platform{OS: "linux", Architecture: "amd64"}
-	if Compatible(Platform{OS: platLocal.OS, Architecture: platLocal.Architecture}, linuxAMD64Goal) {
-		linuxAMD64Goal.Variant = platLocal.Variant
-		linuxAMD64Goal.OSVersion = platLocal.OSVersion
-	}
-	darwinAMD64Goal := Platform{OS: "darwin", Architecture: "amd64"}
-	if Compatible(Platform{OS: platLocal.OS, Architecture: platLocal.Architecture}, darwinAMD64Goal) {
-		darwinAMD64Goal.Variant = platLocal.Variant
-		darwinAMD64Goal.OSVersion = platLocal.OSVersion
-	}
-	darwinARM64Goal := Platform{OS: "darwin", Architecture: "arm64"}
-	if Compatible(Platform{OS: platLocal.OS, Architecture: platLocal.Architecture}, darwinARM64Goal) {
-		darwinARM64Goal.Variant = platLocal.Variant
-		darwinARM64Goal.OSVersion = platLocal.OSVersion
+	darwinGoal := Platform{OS: "darwin"}
+	if Compatible(Platform{OS: platLocal.OS}, darwinGoal) {
+		darwinGoal.Architecture = platLocal.Architecture
+		darwinGoal.Variant = platLocal.Variant
+		darwinGoal.OSVersion = platLocal.OSVersion
 	}
 	windowsAMD64Goal := Platform{OS: "windows", Architecture: "amd64"}
 	if Compatible(Platform{OS: platLocal.OS, Architecture: platLocal.Architecture}, windowsAMD64Goal) {
@@ -41,8 +32,8 @@ func TestPlatformParse(t *testing.T) {
 		windowsAMD64Goal.OSVersion = platLocal.OSVersion
 	}
 	windowsAMD64v2Goal := Platform{OS: "windows", Architecture: "amd64", Variant: "v2"}
-	if Compatible(Platform{OS: platLocal.OS, Architecture: platLocal.Architecture}, windowsAMD64Goal) {
-		windowsAMD64Goal.OSVersion = platLocal.OSVersion
+	if Compatible(Platform{OS: platLocal.OS, Architecture: platLocal.Architecture}, windowsAMD64v2Goal) {
+		windowsAMD64v2Goal.OSVersion = platLocal.OSVersion
 	}
 	localAMD64Goal := Platform{OS: platLocal.OS, Architecture: "amd64"}
 	if Compatible(Platform{OS: platLocal.OS, Architecture: platLocal.Architecture}, localAMD64Goal) {
@@ -136,14 +127,19 @@ func TestPlatformParse(t *testing.T) {
 			goal:  linuxGoal,
 		},
 		{
+			name:  "darwin",
+			parse: "darwin",
+			goal:  darwinGoal,
+		},
+		{
 			name:  "macos amd64",
 			parse: "macos/amd64",
-			goal:  darwinAMD64Goal,
+			goal:  Platform{OS: "darwin", Architecture: "amd64"},
 		},
 		{
 			name:  "darwin arm64",
 			parse: "darwin/arm64",
-			goal:  darwinARM64Goal,
+			goal:  Platform{OS: "darwin", Architecture: "arm64"},
 		},
 		{
 			name:  "windows amd64 with version",

